### PR TITLE
ref(ts): Convert issue list searchbar to typescript

### PR DESCRIPTION
**Pull request by scttcper**

### CODE REVIEW

File: src/sentry/static/sentry/app/views/issueList/searchBar.tsx [hunk 1]
```diff
@@ -43,16 +45,33 @@ const SEARCH_ITEMS = [
...
+};
+
+type State = {
+  defaultSearchItems: [SearchItem[], SearchItem[]];
+  recentSearches: string[];
+};
+
+class IssueListSearchBar extends React.Component<Props, State> {
   static propTypes = {
-    ...SmartSearchBar.propTypes,
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/issueList/searchBar.tsx:48]: This doesn't seem to exist

File: src/sentry/static/sentry/app/views/issueList/searchBar.tsx [hunk 2]
```diff
@@ -43,16 +45,33 @@ const SEARCH_ITEMS = [
...
+type Props = React.ComponentProps<typeof SmartSearchBar> & {
+  api: Client;
+  organization: Organization;
+
+  projectIds?: string[];
+  tagValueLoader: (
+    key: string,
+    search: string,
+    projectIds?: string[]
+  ) => Promise<TagValue[]>;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/issueList/searchBar.tsx:57]:
```suggestion
  tagValueLoader: (
    key: string,
    search: string,
    projectIds?: string[]
  ) => Promise<TagValue[]>;
   projectIds?: string[];
```

File: src/sentry/static/sentry/app/views/issueList/searchBar.tsx [hunk 3]
```diff
@@ -43,16 +45,33 @@ const SEARCH_ITEMS = [
   },
 ];
 
-class IssueListSearchBar extends React.Component {
+type Props = React.ComponentProps<typeof SmartSearchBar> & {
+  api: Client;
+  organization: Organization;
+
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/issueList/searchBar.tsx:51]:
```suggestion
```

**Review by priscilawebdev — APPROVED** [review]
I left a few suggestions, but it looks good 😉

**Review by markstory — APPROVED** [review]
Looks good to me with the suggested changes applied.

**Review by k-fish — APPROVED** [review]

**Pull request merged**